vendors/vendor/boards/board/ports/ota_pal_for_aws/ota_pal.h

```diff
@@ -0,0 +1,231 @@
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/**
+ * @file  ota_pal.h
+ * @brief Function declarations for the functions in ota_pal.c
+ */
+
+#ifndef OTA_PAL_H_
+#define OTA_PAL_H_
+
+#include "ota.h"
+
+/**
+ * @brief Abort an OTA transfer.
+ *
+ * Aborts access to an existing open file represented by the OTA file context pFileContext. This is
+ * only valid for jobs that started successfully.
+ *
+ * @note The input OtaFileContext_t pFileContext is checked for NULL by the OTA agent before this
+ * function is called.
+ *
+ * This function may be called before the file is opened, so the file pointer pFileContext->fileHandle
+ * may be NULL when this function is called.
+ *
+ * The file pointer will be set to NULL after this function returns.
+ *
+ * @param[in] pFileContext OTA file context information.
+ *
+ * @return The OtaPalStatus_t error code is a combination of the main OTA PAL interface error and
+ *         the MCU specific sub error code. See ota_platform_interface.h for the OtaPalMainStatus_t
+ *         error codes and your specific PAL implementation for the sub error code.
+ *
+ * Major error codes returned are:
+ *
+ *   OtaPalSuccess: Aborting access to the open file was successful.
+ *   OtaPalFileAbort: Aborting access to the open file context was unsuccessful.
+ */
+OtaPalStatus_t otaPal_Abort( OtaFileContext_t * const pFileContext );
+
+/**
+ * @brief Create a new receive file for the data chunks as they come in.
+ *
+ * @note Opens the file indicated in the OTA file context in the MCU file system.
+ *
+ * @note The previous image may be present in the designated image download partition or file, so the
+ * partition or file must be completely erased or overwritten in this routine.
+ *
+ * @note The input OtaFileContext_t pFileContext is checked for NULL by the OTA agent before this
+ * function is called.
+ * The device file path is a required field in the OTA job document, so pFileContext->pFilePath is
+ * checked for NULL by the OTA agent before this function is called.
+ *
+ * @param[in] pFileContext OTA file context information.
+ *
+ * @return The OtaPalStatus_t error code is a combination of the main OTA PAL interface error and
+ *         the MCU specific sub error code. See ota_platform_interface.h for the OtaPalMainStatus_t
+ *         error codes and your specific PAL implementation for the sub error code.
+ *
+ * Major error codes returned are:
+ *
+ *   OtaPalSuccess: File creation was successful.
+ *   OtaPalRxFileTooLarge: The OTA receive file is too big for the platform to support.
+ *   OtaPalBootInfoCreateFailed: The bootloader information file creation failed.
+ *   OtaPalRxFileCreateFailed: Returned for other errors creating the file in the device's non-volatile memory.
```

Comment:
Mention in case of general error the sub error code must be properly set.

Reply:
I can add a note about this to the OtaPalRxFileCreateFailed step. Isn't it the case though where we always want the user to set this sub error code if they can and it makes sense?